### PR TITLE
feat(events): org created/invited

### PR DESCRIPTION
**Pull request by migmartri**
new set of events for org management

- orgCreate
- orgInvite
- OrgJoined
- OrgLeft

image

This PR also adds some logic to allow `event publishers` to define if the event must contain actor or not. That way the system will not send events that do not have the full context

refs #1628

### CODE REVIEW

File: app/controlplane/pkg/auditor/events/organization.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	_ auditor.LogEntry = (*OrgUserLeft)(nil)
+	_ auditor.LogEntry = (*OrgCreated)(nil)
+)
+
+const (
+	OrgType                    auditor.TargetType = "Organization"
+	userJoinedOrgActionType    string             = "UserJoined"
+	userLeftOrgActionType      string             = "UserLeft"
+	userInvitedToOrgActionType string             = "UserInvited"
+	orgCreatedActionType       string             = "OrgCreated"
```
Comment (javirln) [inline app/controlplane/pkg/auditor/events/organization.go:38]: What do you think about using the full name of the resource rather a shorter version? So `OrganizationCreated` ?

**Review by javirln — APPROVED** [review]
Small comment

File: app/controlplane/pkg/auditor/events/organization.go (hunk 1 above)
Comment (migmartri) [reply app/controlplane/pkg/auditor/events/organization.go:38]: Yeah, you are probably right, let me revisit it

File: app/controlplane/pkg/auditor/events/organization.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+func (p *OrgUserLeft) ActionType() string {
+	return userLeftOrgActionType
+}
+
+func (p *OrgUserLeft) Description() string {
+	return fmt.Sprintf("{{ .ActorEmail }} has left the organization %s", p.OrgName)
+}
+
+// user got invited to the organization
+type OrgUserInvited struct {
```
[the remark this answers is not in the document]
Comment (migmartri) [reply app/controlplane/pkg/auditor/events/organization.go:106]: it is the actor, see the description of the evetn

File: app/controlplane/pkg/auditor/logentry.go [hunk 3]
```diff
@@ -55,6 +55,7 @@ type LogEntry interface {
 	TargetID() *uuid.UUID
 	// Description returns a templatable string, see the DescriptionVariables struct.
 	Description() string
+	RequiresActor() bool
```
Comment (javirln) [inline app/controlplane/pkg/auditor/logentry.go:58]: Ah ok I haven't seen this!

**Review by jiparis — APPROVED** [review]

**Pull request merged**